efficiencyFactors.py

```diff
@@ -940,7 +938,11 @@ def geteffs(channel, SampleList, m4l_bins, m4l_low, m4l_high, obs_reco, obs_gen,
 if (not opt.CHAN==''):
     ext='_'+opt.CHAN
 
-output_file_name = datacardInputs+'/inputs_sig_'+label+ext+'.py'
+
+if not os.path.isdir(opt.ERA): os.mkdir(opt.ERA)
+if not os.path.isdir(opt.ERA+"/"+datacardInputs): os.mkdir(opt.ERA+"/"+datacardInputs)
+
+output_file_name = opt.ERA+"/"+datacardInputs+'/inputs_sig_'+label+ext+'.py'
```

Comment:
Here, it will create directory like:

```bash
2016/datacardInputs/,
2017/datacardInputs/,
```

What do you think about reversing the order, like below?

```bash
datacardInputs/2016/
datacardInputs/2017/
```

Then we will have only one main directory having sub-directories.

Reply:
I'd be in favour of doing it the other way around as it keeps the similar structure to before just adding separate folders (easy to propagate) + you can just pick all the inputs for one era by taking a look into the ```year``` folder.